File: Day 1/day1.py
nums = [10, 15, 3, 7]
k = 17

def compute(nums, k):
    # sort the list of numbers O(nlogn)
    nums.sort()

    # Set L and R Pointers
    l = len(nums) - 1
    r = 0

    # Loop through num list to determine if there is a valid answer
    # Runtime: O(n)
    while (r < l):
        if nums[l] + nums[r] == k:
            return True
        elif nums[l] + nums[r] < k:
            r += 1
        else:
            l -= 1
    return False

def main():
    print(compute(nums, k))

if __name__ == "__main__":
    main()

File: Day 1/day1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool compute(vector<int>nums, int k) {
    std::sort(nums.begin(), nums.end());
    
    int r = 0;
    int l = nums.size() - 1;
    while (r < l) {
        if (nums[r] + nums[l] == k) {
            return true;
        } else if (nums[r] + nums[l] < k) {
            r += 1;
        } else {
            l -= 1;
        }
    }
    return false;
}

int main() {
    vector<int> numbers = {1, 2, 3, 4, 5};
    int target = 6;
    cout << compute(numbers, target) << endl;
    return 0;
}